contracts/UsulVetoGuard.sol

```diff
@@ -0,0 +1,153 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+import "./interfaces/IUsulVetoGuard.sol";
+import "./interfaces/IVetoVoting.sol";
+import "./interfaces/IOZLinearVoting.sol";
+import "./interfaces/IUsul.sol";
+import "./TransactionHasher.sol";
+import "./FractalBaseGuard.sol";
+import "@gnosis.pm/zodiac/contracts/factory/FactoryFriendly.sol";
+import "@gnosis.pm/safe-contracts/contracts/common/Enum.sol";
+
+/// @notice A guard contract that prevents transactions that have been vetoed from being executed a Gnosis Safe
+/// @notice through a Usul module with an attached OZLinearVoting strategy
+contract UsulVetoGuard is
+    IUsulVetoGuard,
+    TransactionHasher,
+    FactoryFriendly,
+    FractalBaseGuard
+{
+    IVetoVoting public vetoVoting;
+    IOZLinearVoting public ozLinearVoting;
+    IUsul public usul;
+    mapping(bytes32 => uint256) transactionQueuedBlock;
+
+    /// @notice Initialize function, will be triggered when a new proxy is deployed
+    /// @param initializeParams Parameters of initialization encoded
+    function setUp(bytes memory initializeParams) public override initializer {
+        __Ownable_init();
+        (
+            address _owner,
+            address _vetoVoting,
+            address _ozLinearVoting,
+            address _usul
+        ) = abi.decode(initializeParams, (address, address, address, address));
+
+        transferOwnership(_owner);
+        vetoVoting = IVetoVoting(_vetoVoting);
+        ozLinearVoting = IOZLinearVoting(_ozLinearVoting);
+        usul = IUsul(_usul);
+
+        emit UsulVetoGuardSetup(
+            msg.sender,
+            _owner,
+            _vetoVoting,
+            _ozLinearVoting,
+            _usul
+        );
+    }
+
+    /// @notice Queues a transaction for execution
+    /// @param proposalId The ID of the proposal to queue
+    function queueProposal(uint256 proposalId) external {
```

Comment:
Yeah I'm confused on how this works exactly.  Does the Usul mod check a list of guards before queueing?

Reply:
Yes, this scenario is accounted for, just pushed up a test for this. Basically, the Usul module calls `checkTransaction` on the UsulVetoGuard every time `execute` is called. If the transaction was not queued on the `UsulVetoGuard`, then it will hit this require statement:

https://github.com/decent-dao/fractal-contracts/blob/usul-veto-guard/contracts/UsulVetoGuard.sol#L103